velox/functions/sparksql/Arithmetic.h

```diff
@@ -483,4 +484,94 @@ struct RIntFunction {
     result = std::rint(input);
   }
 };
+
+template <typename TExec>
+struct CheckedAddFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+  template <typename T>
+  FOLLY_ALWAYS_INLINE Status call(T& result, const T& a, const T& b) {
+    if constexpr (std::is_integral_v<T>) {
+      T res;
+      bool overflow = __builtin_add_overflow(a, b, &res);
+      if (UNLIKELY(overflow)) {
+        if (threadSkipErrorDetails()) {
+          return Status::UserError();
+        }
+        return Status::UserError(
+            "[ARITHMETIC_OVERFLOW] overflow: {} + {}", a, b);
+      }
+      result = res;
+    } else {
+      result = a + b;
```

Comment:
Does this imply that overflow is allowed for non-integer types?

Reply:
Yes, I think for Float and Double there is no overflow. Inf + 1 still equals Inf.

Reply:
For non-integer types, would it be better to return first to improve readability?  
>if constexpr (!std::is_integral_v<T>) {
  result = a + b;
  return Status::OK();
}

Reply:
We need this check wrap the function` __builtin_XX_overflow`, otherwise we get the compilation error:
`/var/git/velox/./velox/functions/sparksql/Arithmetic.h:540:43: error: argument 1 in call to function ‘__builtin_mul_overflow’ does not have integral type`